src/ResourceManager/RecoveryServices/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/A2A/AsrA2ATests.ps1

```diff
@@ -0,0 +1,146 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+########################## Site Recovery Tests #############################
+
+##Default Value ##
+
+<#
+.SYNOPSIS
+    NewA2ADiskReplicationConfiguration creation test.
+#>
+
+function Test-NewA2ADiskReplicationConfiguration
+{
+    $recoveryStorageAccountId ="/subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/a2a-rg/providers/Microsoft.Storage/storageAccounts/a2argdisks412"
+    $logStorageAccountId = "/subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/ltrgp1705152333/providers/Microsoft.Storage/storageAccounts/stagingsa2name1705152333"
+    $vhdUri = "https://powershelltestdiag414.blob.core.windows.net/vhds/pslinV2-520180112143232.vhd"
+
+    $v = New-AzureRmRecoveryServicesAsrAzureToAzureDiskReplicationConfig -VhdUri  $vhdUri `
+        -RecoveryAzureStorageAccountId $recoveryStorageAccountId `
+        -LogStorageAccountId   $logStorageAccountId
+
+     Assert-True { $v.vhdUri -eq $vhdUri }
+     Assert-True { $v.recoveryAzureStorageAccountId -eq $recoveryStorageAccountId  }
+     Assert-True { $v.logStorageAccountId -eq $logStorageAccountId }
+}
+
+
+<#
+.SYNOPSIS
+    NewA2ADiskReplicationConfiguration creation test.
+#>
+function Test-NewA2AManagedDiskReplicationConfiguration
+{
+    $logStorageAccountId = "fdd"
+    $DiskId = "diskId"
+    $RecoveryResourceGroupId = "3"
+    $RecoveryReplicaDiskAccountType = "Premium_LRS"
+    $RecoveryTargetDiskAccountType = "Premium_LRS"
+
+    $v = New-AzureRmRecoveryServicesAsrAzureToAzureDiskReplicationConfig -managed -LogStorageAccountId $logStorageAccountId `
+         -DiskId "diskId" -RecoveryResourceGroupId  $RecoveryResourceGroupId -RecoveryReplicaDiskAccountType  $RecoveryReplicaDiskAccountType `
+         -RecoveryTargetDiskAccountType $RecoveryTargetDiskAccountType
+
+     Assert-True { $v.LogStorageAccountId -eq $LogStorageAccountId }
+     Assert-True { $v.DiskId -eq $DiskId  }
+     Assert-True { $v.RecoveryResourceGroupId -eq $RecoveryResourceGroupId }
+}
+
+<#
+.SYNOPSIS 
+    Test GetAsrFabric new parametersets
+#>
+function Test-NewAsrFabric {
+    $seed = 35;
+        $vaultRgLocation = getVaultRgLocation
+        $vaultName = getVaultName
+        $vaultLocation = getVaultLocation
+        $vaultRg = getVaultRg
+        $primaryLocation = getPrimaryLocation
+        $recoveryLocation = getRecoveryLocation
+        $primaryFabricName = getPrimaryFabric
+        $recoveryFabricName = getRecoveryFabric
+        
+        New-AzureRmResourceGroup -name $vaultRg -location $vaultRgLocation -force
+        [Microsoft.Azure.Test.TestUtilities]::Wait(20 * 1000)
+    # vault Creation
+        New-azureRmRecoveryServicesVault -ResourceGroupName $vaultRg -Name $vaultName -Location $vaultLocation
+        [Microsoft.Azure.Test.TestUtilities]::Wait(20 * 1000)
```

Comment:
Should be using `[Microsoft.Rest.ClientRuntime.Azure.TestFramework.TestUtilities]::Wait`